src/concrete/ml/onnx/ops_impl.py

```diff
@@ -897,7 +899,40 @@ def numpy_equal(
     Returns:
         Tuple[numpy.ndarray]: Output tensor
     """
+    return (numpy.equal(x, y),)
+
+
+def rounded_numpy_equal_for_trees(
+    x: numpy.ndarray,
+    y: numpy.ndarray,
+    *,
+    lsbs_to_remove_for_trees: Optional[int] = None,
+) -> Tuple[numpy.ndarray]:
+    """Compute rounded equal in numpy according to ONNX spec for tree-based models only.
+
+    See https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Equal-11
+
+    Args:
+        x (numpy.ndarray): Input tensor
+        y (numpy.ndarray): Input tensor
+        lsbs_to_remove_for_trees (Optional[int]): Number of the least significant bits to remove
+            for tree-based models only.
+
+    Returns:
+        Tuple[numpy.ndarray]: Output tensor
+    """
+
+    # For tree-based models in the second stage, x == y is equivalent to x <= y or x < y - 1
+    # Because y is the max sum, see this paper: https://arxiv.org/pdf/2010.04804.pdf
+    # The approach x <= y, is equivalent to: x - y <= 0 or y - x >= 0
+    # We take y - x >= 0, because with `rounding_bit_pattern` feature, it gives accurate outputs
```

Comment:
thanks ! Still, `it gives accurate outputs compared to x - y <= 0` is a bit unclear. I thin what you mean is just that it is not symmetric/equivalent right (because of half) ? I would maybe be more precise here. Also, do not forget to also put this message in the other operators !